src/_guides/language/effective-dart/design.md

```diff
@@ -1219,103 +1204,90 @@ var names = people.map((person) => person.name);
-    }
-    {% endprettify %}
-
-    (We call this *downwards inference* because we think of the code as a tree,
-    and the initializer expression is a subtree of the larger declaration tree.)
-
-*   You can annotate the generic invocation expression in the initializer, which
-    inference then uses to infer the field or variable's type:
-
-    {:.bad-style}
-    <?code-excerpt "misc/lib/effective_dart/design_bad.dart (explicit-field)"?>
-    {% prettify dart %}
-    class Histogram {
-      final counts = <String, int>{};
-    }
-    {% endprettify %}
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (redundant-explicit)"?>
+{% prettify dart %}
+var things = new Set<String>();
+{% endprettify %}
 
-    (Conversely, we call this *upwards inference* because the type is flowing up
-    from the initializer expression to the surrounding declaration.)
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (redundant-explicit)"?>
+{% prettify dart %}
+var things = new Set();
+{% endprettify %}
 
-In other words, you can put the type "on the left" or "on the right". Since we
-already have a guideline encouraging you to explicitly type top-level
-declarations, this guideline just restates that you should prefer typing "on the
-left".
+Here, since the variable has no type annotation, there isn't enough context to
+determine what kind of Set to create, so the type argument should be provided
+explicitly.
 
 
-### PREFER type annotating generic invocations for local variables that can't be inferred.
+### DO annotate when Dart infers the wrong type.
 
-This is the other side of the previous rule. For *local* variables, the guidance
-is that you should avoid annotating them when possible. To stay consistent with
-that, if inference fails on a local variable, prefer fixing that by typing the
-generic invocation so that you don't need to annotate the local variable itself:
+Sometimes, Dart infers a type, but not the type you want. For example, you may
+want a variable's type to be a supertype of the initializer's type so that you
+can later assign some other sibling type to the variable:
 
 {:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (explicit-local)"?>
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (inferred-wrong)"?>
 {% prettify dart %}
-void countOccurrences(List<String> words) {
-  var wordCounts = <String, int>{};
-  // ...
+num highScore(List<num> scores) {
+  num highest = 0;
+  for (var score in scores) {
+    if (score > highest) highest = score;
+  }
+  return highest;
 }
 {% endprettify %}
 
-Of course, if the initializer isn't a generic invocation or there isn't a way to
-put a type annotation in the initializer to get inference working, then typing
-the local variable itself is the way to go.
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (inferred-wrong)" replace="/ +\/\/ ignore: invalid_assignment\n//g"?>
```

Comment:
You can probably shorten the `replace` expression:

    replace="/ +\/\/ ignore: .*?\n//g"

Reply:
Done.